File: nodejs/english-demo/service/httpclient.js
var rp = require('request-promise');
var model = require('../model/model.js');

/**
 * send http request with json body
 *
 * @param url 
 * @param jsonStr json string
 * 
 */
function httpPostJson(url, json){
    return new Promise(function(resolve){
        req = {
            url: url,
            method: "POST",
            json: true,
            encoding: 'utf-8',
            headers: {
                "content-type": "application/json",
            },
            body: json
        }


        rp(req).then(function(parsedBody) {
            var resp = new model.XPYunResp();
            resp.httpStatusCode = 200;

            resp.content = parsedBody
 
            resolve(resp);      
        }).catch(function(err) {
            console.log(`[Error]: ${err}`)
            resolve(null);
        }); 
    });
}


module.exports = {httpPostJson};

File: go/english/src/xpyun-opensdk/service/httpclient.go
package service

import (
	"bytes"
	"encoding/json"
	"fmt"
	"io/ioutil"
	"net/http"
	"xpyun-opensdk/model"
)

/**
 * send http request with json body
 *
 * @param url
 * @param jsonStr
 *
 */
func HttpPostJson(url string, data interface{}) *model.XPYunResp {
	b, _ := json.Marshal(data)

	fmt.Println(string(b))

	resp, err := http.Post(url, "application/json", bytes.NewBuffer(b))
	if err != nil {
		fmt.Println(err)
	}
	defer resp.Body.Close()
	body, err := ioutil.ReadAll(resp.Body)
	if err != nil {
		fmt.Println(fmt.Sprintf("post json error:%+v", err))
	}

	result := model.XPYunResp{
		HttpStatusCode: resp.StatusCode,
	}

	var content model.XPYunRespContent
	err = json.Unmarshal(body, &content)
	if err == nil {
		result.Content = &content
	} else {
		fmt.Println(fmt.Sprintf("unmarshal body failed, error:%+v", err))
	}

	return &result
}
